### PR TITLE
Updates `ndpyramid` to use the newer version of Xarray that integrated Datatree. 

### LINKED ISSUE #151: Cannot import ndpyramid v0.3.1 with xarray v2024.10.0
ndpyramid v0.3.1
xarray v2024.10.0

MRE:  ```import ndpyramid```

```python
ImportError                               Traceback (most recent call last)
Cell In[7], line 1
----> 1 import ndpyramid

~/src/examples/multichannel_timeseries/envs/default/lib/python3.11/site-packages/ndpyramid/__init__.py:3      1 # flake8: noqa
----> 3 from .create import pyramid_create
      4 from .coarsen import pyramid_coarsen
      5 from .reproject import pyramid_reproject

~/src/examples/multichannel_timeseries/envs/default/lib/python3.11/site-packages/ndpyramid/create.py:5      1 from __future__ import annotations  # noqa: F401
      3 from typing import Callable
----> 5 import datatree as dt
      6 import xarray as xr
      8 from .utils import get_version, multiscales_template

~/src/examples/multichannel_timeseries/envs/default/lib/python3.11/site-packages/datatree/__init__.py:2
      1 # import public API
----> 2 from .datatree import DataTree
      3 from .extensions import register_datatree_accessor
      4 from .io import open_datatree

~/src/examples/multichannel_timeseries/envs/default/lib/python3.11/site-packages/datatree/datatree.py:33
     31 from xarray.core.merge import dataset_update_method
     32 from xarray.core.options import OPTIONS as XR_OPTS
---> 33 from xarray.core.utils import (
     34     Default,
     35     Frozen,
     36     HybridMappingProxy,
     37     _default,
     38     either_dict_or_kwargs,
     39     maybe_wrap_array,
     40 )
     41 from xarray.core.variable import Variable
     43 from . import formatting, formatting_html

ImportError: cannot import name 'HybridMappingProxy' from 'xarray.core.utils'
```

### PR DESCRIPTION
**Pull request by norlandrhagen**
- fixes #151

### CODE REVIEW

File: .github/workflows/pypi-release.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
       - uses: actions/setup-python@v5
         name: Install Python
         with:
-          python-version: 3.9
+          python-version: 3.10
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:54]:
```suggestion
          python-version: '3.10'
```

File: .github/workflows/pypi-release.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ jobs:
...
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:19]:
```suggestion
          python-version: '3.10'
```

File: pyproject.toml [hunk 3]
```diff
@@ -86,13 +87,13 @@ write_to_template = '__version__ = "{version}"'
 
 [tool.black]
 line-length = 100
-target-version = ['py39']
+target-version = ['py312']
```
Comment (andersy005) [inline pyproject.toml:90]: we should probably target the minimum supported version instead

```suggestion
target-version = ['py310']
```

**Comment by norlandrhagen** [comment]
Thanks for the packaging/dist help @andersy005! Any other things we should address with this PR?

**Pull request merged**